lib/functions/ProofUtil.ts

```diff
@@ -35,13 +53,23 @@ function partiallyValidateJWS(jws: string): void {
   }
 }
 
-function setJWSDefaults(args: JWTSignerArgs, issuerUrl: string, clientId?: string): JWTSignerArgs {
+function setJWSDefaults(
+  args: ProofOfPossessionCallbackArgs,
+  endpointMetadata: EndpointMetadata,
+  clientId?: string
+): { header: JWTHeader; payload: JWTPayload } {
   const now = +new Date();
-  const aud = args.payload.aud ? args.payload.aud : issuerUrl;
+  if (!endpointMetadata) {
+    throw new Error('No endpointMetadata provided');
+  }
+  const aud = endpointMetadata.issuer ? endpointMetadata.issuer : args.payload.aud;
   if (!aud) {
     throw new Error('No issuer url provided');
   }
-  const iss = args.payload.iss ? args.payload.iss : clientId;
+  if (!args.kid) {
+    throw new Error('No kid provided');
+  }
+  const iss = clientId ? clientId : args.payload.iss;
```

Comment:
The iss value should be fully optional and only populated if we actually do use a clientId

Reply:
populated after the creation of `jwtArgs` and _if present_